generator/config/expression/substrBytes.yaml

```diff
@@ -23,3 +23,25 @@ arguments:
             - resolvesToInt
         description: |
             If length is a negative number, $substr returns a substring that starts at the specified index and includes the rest of the string.
+tests:
+    -
+        name: 'Single-Byte Character Set'
+        link: 'https://www.mongodb.com/docs/manual/reference/operator/aggregation/substrBytes/#single-byte-character-set'
+        pipeline:
+            -
+                $project:
+                    item: 1
+                    yearSubstring:
+                        $substrBytes:
+                            - '$quarter'
+                            - 0
+                            - 2
+                    quarterSubtring:
+                        $substrBytes:
+                            - '$quarter'
+                            - 2
+                            -
+                                $subtract:
+                                    -
+                                        $strLenBytes: '$quarter'
+                                    - 2
```

Comment:
Is the extra indentation the result of a script parsing the example code? If not, I think this can be more concise.

You can inline `{ $strLenBytes: '$quarter' }`, and `$subtract` shouldn't need a new line and indentation. We frequently do the following in spec tests:

```yaml
runOnRequirements:
  - minServerVersion: "4.0"
    topologies: [ replicaset ]
```

The same may apply to other files but I'll defer to you if it's worth addressing. I personally find the dashes for array elements on their own line a bit confusing when visually scanning a file.

Reply:
I'm using a very standardized and exploded syntax for Yaml. It's generated by my `js2yaml.html` tool.

I can reindent all the Yaml with 2 spaces and remove the empty `-` lines in an other PR. PHPStorm can do this formatting for me.

Reply:
Entirely up to you. This is the kind of formatting I try to avoid when writing spec tests, but I understand why a tool might produce it.